lang/src/accounts/account.rs

```diff
@@ -243,12 +241,12 @@ impl<'a, T: AccountSerialize + AccountDeserialize + Owner + Clone> Account<'a, T
 
     /// Deserializes the given `info` into a `Account`.
     #[inline(never)]
-    pub fn try_from(info: &AccountInfo<'a>) -> Result<Account<'a, T>, ProgramError> {
+    pub fn try_from(info: &AccountInfo<'a>) -> AnchorResult<Account<'a, T>> {
```

Comment:
Idea: when invoking these functions from the derive(Accounts) macro, we probably want to map the error and use the field span to display the line number + file of the accounts context field.

That is, if we have a struct like

```
#[derive(Accounts)]
pub struct MyAccounts {
  program: Account<'info, Data>, 
}
```

Then the error should point to the `program` field above. Instead of this file.

Reply:
yes, thats the plan!

this is by no means close to done Ive been busy making the most basic of programs compile